src/content/en/fundamentals/security/credential-management/retrieve-credentials.md

```diff
@@ -4,369 +4,396 @@ book_path: /web/fundamentals/_book.yaml
-      </td>
-      <td>
-        <code>Boolean</code><br>
-        Set to <code>true</code> to avoid showing account chooser UI.
-      </td>
-    </tr>
-  </tbody>
-</table>
-
-## Get a credential
-
-### Get a credential automatically
-
-To sign a user in automatically, request a credential object with
-`unmediated: true`, as soon as they land on your website, for example:
-
-<pre class="prettyprint">
-navigator.credentials.get({
-  password: true,
-  <strong>unmediated: true,</strong> // request a credential without user mediation
-  federated: {
-    providers: [
-      'https://account.google.com',
-      'https://www.facebook.com'
-    ]
-  }
-})
-</pre>
-
-<figure class="attempt-right">
-  <img src="imgs/auto-sign-in.png">
-  <figcaption>Notification for auto signed-in user</figcaption>
-</figure>
-
-This request resolves immediately with a credential object and won't display
-an account chooser. When the browser obtains credential information,
-a notification pops up:
-
-<div class="clearfix"></div>
-
-
-### Get a credential via the account chooser
+    }
 
-<figure class="attempt-right">
-  <img src="imgs/account-chooser.png">
-  <figcaption>Account chooser UI</figcaption>
-</figure>
+When the `navigator.credentials.get()` resolves,
+it returns either undefined or a credential object.
+To determine whether it is a `PasswordCredential` or a `FederatedCredential`,
+simply look at the `.type` property of the object,
+which will be either `password` or `federated`.
+
+If the `.type` is `federated`,
+the `.provider` property is a string that represents the identity provider.
+
+### Authenticate user
+
+Once you have the credential,
+run an authentication flow depending on the type of credential,
+`password` or `federated`:
+
+    }).then(c => {
+     if (c) {
+       switch (c.type) {
+         case 'password':
+           return sendRequest(c);
+           break;
+         case 'federated':
+           return gSignIn(c);
+           break;
+       }
+     } else {
+       return Promise.resolve();
+     }
+
+When the function resolves,
+check if you've received a credential object.
+If not, it means auto sign-in couldn’t happen.
+Silently dismiss the auto sign-in process.
+
+### Update UI
+
+If the authentication is successful,
+update the UI or forward the user to the personalized page:
+
+    }).then(profile => {
+     if (profile) {
+       updateUI(profile);
+     }
+
+### Don’t forget to show authentication error message
+
+One important tip: if you succeed in obtaining a credential object 
+but fail to authenticate the user, you should show an error message:
+
+        }).catch(error => {
+          showError('Sign-in Failed');
+        });
+      }
+    }
 
-If a user requires mediation, or has multiple accounts, use the account
-chooser to let the user sign-in, skipping the ordinary sign-in form.
+To avoid user confusion,
+users should see a blue toast saying “Signing in”
+at the time of getting the credential object:
```

Comment:
I moved it above the sentence introducing the code snippet, as it is the image showing the sign in before failure, and the failure logically follows on from this.